src/NetSparkle/SparkleUpdater.cs

```diff
@@ -428,7 +431,14 @@ public IAppCastHandler AppCastHandler
             set => _appCastHandler = value;
         }
 
-#endregion
+        /// <summary>
+        /// The custom appcast filter instance - can also be null; usually used to implement a downgrade strategy
+        /// <seealso cref="IAppCastFilter"/>
+        /// </summary>
+        /// <remarks>Note: calls to methods on this interface are made from background threads - dont access UI objects from within this method</remarks>
+        public IAppCastFilter AppCastFilter { get => _appCastFilter; set => _appCastFilter = value; }
```

Comment:
You never pass this to the `XMLAppCast`, so this property does nothing at the moment for the user. I think you can just remove this from the `SparkleUpdater` object entirely and let users set this directly on their `XMLAppCast` object, now that I think about it, thus this PR would make no changes to `SparkleUpdater`.

Reply:
In other words, just init your `SparkleUpdater` with something like this to avoid having to do any casts later on:

```csharp
var updater = new SparkleUpdater(...)
{
    AppCastHandler = new XMLAppCast() { AppCastFilter = this }
};
```

Reply:
I knew I was missing *something* there.   

Thank you.  That tidies this up nicely; although I'd still consider it implementation black magic - which could I admit be solved by a decent application of documentation that I'd be more than happy to write up.  

I have yet to test this but my code now looks cleaner... thank you!

```csharp
public sealed class VersionUpdater : ILogger, IAppCastFilter
```
...
```csharp
public VersionUpdater(string appCastUrl, string publicKey, IUIFactory uiFactory, bool filterOutBetaItems, bool forceDowngradeLatest)
    {
        Updater = new SparkleUpdater(appCastUrl,
            new Ed25519Checker(SecurityMode.Strict, publicKey), null, uiFactory,
            this)
        {
            AppCastFilter = this
        };

        AppCastUrl = appCastUrl;
        FilterOutBetaItems = filterOutBetaItems;
        ForceDowngradeLatestVersion = forceDowngradeLatest;
        Updater.LogWriter = this;
    }
```

In my case I throw the entirely SparkleUpdater instance away every time the user causes a switch between stable / beta channel.

Reply:
My suggestion sets the filter right into the `XMLAppCast`, not the `SparkleUpdater`, which allows you to set it when you create the initial `SparkleUpdater` object. If it was set on the `SparkleUpdater` (how you had it before), it would be:

```csharp
var updater = new SparkleUpdater(...)
{
    AppCastHandler = new XMLAppCast(),
    AppCastFilter = this,
};
```

Reply:
I think based on your comment below in the main thread that you're on the same page. My bad.